config/routes.rb

```diff
@@ -556,7 +556,16 @@ def draw(routes_name)
                         path: 'items',
                         as: :items
             end
-            resources :projects, only: %i(index show) do
+            resources :projects, only: %i(index show), path: 'projects',
+                      as: :projects do
+              resources :user_projects, only: %i(index show),
+                path: 'user_projects', as: :user_projects
+              resources :project_comments, only: %i(index show),
+                path: 'project_comments', as: :project_comments
+              resources :reports, only: %i(index show),
+                path: 'reports', as: :reports
+              get 'activities', to: 'activities#project_activities'
+              get 'activities/:id', to: 'activities#project_activity'
```

Comment:
I think generally, the "view individual activity" endpoint should be refactored into its own, individual endpoint - practically, you are querying the same "individual activity" endpoint from various parts of the hierarchy, which means this endpoint should be located somewhere at the root, e.g. `api/v1/activities`, or `api/v1/teams/:team_id/activities` at the most. This would also cut down the "duplicity", e.g. now there are already 2 endpoints for viewing individual activity:
```
/api/v1/teams/:team_id/projects/:project_id/experiments/:experiment_id/tasks/:task_id/activities/:id
/api/v1/teams/:team_id/projects/:project_id/activities/:id
```

I think you can refactor this in scope of this task.